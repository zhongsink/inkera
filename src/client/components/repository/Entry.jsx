import React from 'react';
import './styles/Entry.less';

const Entry = ({ repository }) => (
  <div className="entry">
    <div className="left">
      <div className="full-name">
        <a href="">{ repository.full_name }</a>
      </div>
      <div className="description">
        { repository.description }
      </div>
      <div className="topics">
        { repository.topics.slice().splice(0, 4).map(topic => <div className="topic" key={topic}>{topic}</div>) }
      </div>
      <div className="update-time">
        { repository.updated_at }
      </div>
    </div>
    <div className="middle">
      <span className="language" />
      <p>{ repository.language }</p>
    </div>
    <div className="right">
      <svg className="start" height="20" role="img" version="1.1" viewBox="0 0 14 16" width="15">
        <path fillRule="evenodd" d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z" />
      </svg>
      <p>
        { repository.stargazers_count }
      </p>
    </div>
  </div>
);

export default Entry;
