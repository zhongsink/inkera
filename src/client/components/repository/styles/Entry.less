@import "~themes/default.less";

.entry {
  border-top-style: solid;
  border-top-color: rgba(200, 200, 200, 0.6);
  border-top-width: 1px;
  display: flex;
  flex-direction: row;
  align-items: flex-start;

  .left {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: left;
    width: 67%;

    .full-name {
      margin-top: 2em;

      a {
        max-width: 19em;
        display: -webkit-box;
        -webkit-box-orient: vertical;
        -webkit-line-clamp: 3;
        overflow: hidden;
        color: #0366d6;
        text-decoration: none;
        font-size: 5/3em;
        font-weight: 600;
      }
    }

    .description {
      .text-normal;
      margin-top: 0.5em;

      max-width: 25em;
      display: -webkit-box;
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 3;
      overflow: hidden;
    } 

    .topics {
      display: flex;
      margin-top: 0.5em;

      .topic {
        .text-normal;

        font-size: 3/4em;
        padding: 0.3em 0.9em;
        margin: 0 1em 0 0;
        color: rgb(3, 102, 214);
        background-color: #f1f8ff;
      }
    }

    .update-time {
      .text-normal;
      margin: 1em 0 1.8em 0;
    }
  }

  .middle {
    width: 17%;
    margin-top: 2.5em;
    display: flex;
    flex-direction: row;
    align-items: center;

    .language {
      background-color: #f1e05a;
      border-radius: 50%;
      width: 1em;
      height: 1em;
    }

    p {
      .text-normal;
      margin: 0 0 0 0.2em;
    }
  }

  .right {
    width: 16%;
    margin-top: 2.5em;
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;

    p {
      .text-normal;
      margin: 0 1em 0 0.2em;
    }
  }
}
