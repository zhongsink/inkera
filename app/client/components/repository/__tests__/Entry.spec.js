import Entry from '../Entry';
import { snapshot } from '../../../../shared/test';

describe('components/repository/Entry', () => {
  snapshot({
    component: Entry,
    name: 'Entry',
    props: {
      repository: {
        full_name: 'full_name',
        description: 'description',
        topics: [],
        update_at: 'update_at',
        language: 'language',
        stargazers_count: 100,
      },
    },
  });
});
