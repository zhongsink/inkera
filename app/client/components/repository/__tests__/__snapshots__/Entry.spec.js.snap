// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/repository/Entry Entry should render correctly 1`] = `
<div
  className="entry"
>
  <div
    className="left"
  >
    <div
      className="full-name"
    >
      <a
        href=""
      >
        full_name
      </a>
    </div>
    <div
      className="description"
    >
      description
    </div>
    <div
      className="topics"
    />
    <div
      className="update-time"
    />
  </div>
  <div
    className="middle"
  >
    <span
      className="language"
    />
    <p>
      language
    </p>
  </div>
  <div
    className="right"
  >
    <svg
      className="start"
      height="20"
      role="img"
      version="1.1"
      viewBox="0 0 14 16"
      width="15"
    >
      <path
        d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"
        fillRule="evenodd"
      />
    </svg>
    <p>
      100
    </p>
  </div>
</div>
`;
